### ISSUE TITLE
Issues with dependencies between jobs running on chunk level

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1341]
In GitLab by @abaer-kit on Jul 5, 2024, 16:47

Hello @dbeltrankyl and @kinow,

### Autosubmit Version

4.1.8

### Summary

If a job (e.g. RUN) depends on its previous chunk (RUN-1) and on another job running on chunk level, all dependencies of the chunks 2+ will be lost.
In the attached plot [a01k_20240705_1640.pdf](https://earth.bsc.es/gitlab/es/autosubmit/uploads/bb6558b7ba3f4891223816474aa3ebd7/a01k_20240705_1640.pdf), I would have expected further dependencies:\
`a01k_2020100100_fc0_2_FILE_CREATOR` -> `a01k_2020100100_fc0_2_RUN`\
`a01k_2020100100_fc0_3_FILE_CREATOR` -> `a01k_2020100100_fc0_3_RUN`

### Steps to reproduce

Create an experiment with the following configuration and custom config in `"%PROJDIR%/basic_conf.yml"`
`basic_conf.yml`:
```yaml
DEFAULT:
  CUSTOM_CONFIG:
    PRE:
      - "%PROJDIR%/dummy_conf.yml"
      - "%PROJDIR%/jobs-chunks.yml"
```
`dummy_conf.yml`:
```yaml
experiment:
  DATELIST: 20201001
  MEMBERS: "fc0"
  CHUNKSIZEUNIT: hour
  CHUNKSIZE: 1
  NUMCHUNKS: 3
  CHUNKINI: 0
  CALENDAR: standard
#
PLATFORMS:
  DUMMY:
    TYPE: ps
    HOST: 'dummy.host'
    USER: 'dummy'
    SCRATCH_DIR: "/home"
```
`jobs-chunks.yml`:
```yaml
JOBS:
  FILE_CHECKER:
    FILE: file_checker.sh
    RUNNING: chunk
  
  FILE_CREATOR:
    FILE: file_creator.sh
    RUNNING: chunk
    DEPENDENCIES:
      FILE_CHECKER:

  RUN:
    FILE: run.sh
    RUNNING: chunk
    DEPENDENCIES:
      RUN-1:
      FILE_CREATOR:
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jul 8, 2024, 09:33

Hello @abaer-kit 

Thanks for the report, I fixed it on master, scheduled for 4.1.9.

I have a regression test to compare the workflow between 4.0 and 4.1 and added this case there.

The rest of the cases in the test still have the same structure as before, so there shouldn't be issues.

**Comment by kinow** [comment]
In GitLab by @abaer-kit on Jul 9, 2024, 10:53

Great, thanks!
